filesystem.go

```diff
@@ -154,7 +154,7 @@ func (fs *filesystem) Logger() *logrus.Entry {
 func (fs *filesystem) createAllResources(pod Pod) (err error) {
 	for _, resource := range []podResource{stateFileType, configFileType} {
 		_, path, _ := fs.podURI(pod.id, resource)
-		err = os.MkdirAll(path, os.ModeDir)
+		err = os.MkdirAll(path, os.ModeDir|dirMode)
```

Comment:
Do we even need os.ModeDir here with the mkdir system call?

Reply:
Yep, as per the Issue and commit. If we don't do that then we get dirs created as `d---------` (01000). `ModeDir` **only** gives you the `d` bit - which is why the go docs say you must have it when creating dirs - but they omit to say you still need to supply the rest of the mode bits you want... afaict. Feel free to double check and verify. When I tested this it did what I expected (changed the dirs from 01000 to 01750)

Reply:
Ah, or maybe I see what you mean now ... :-) We might only need `dirMode`, like we use for other calls to `os.Mkdir[All]`.

From the golang docs at:
https://golang.org/pkg/os/#FileMode
we have:

> The only required bit is ModeDir for directories.

But, at https://github.com/containers/virtcontainers/blob/master/filesystem.go#L91
we have
```
const dirMode = os.FileMode(0750)
```
Maybe the better solution then is to add `os.ModeDir` to our `dirMode`, and just use that everywhere (as we pretty much do already).

thoughts @amshinde @sboeuf ?

Reply:
@grahamwhaley the point from @amshinde is that we might not need `os.ModeDir` since this should be added by default by `Mkdir()`.
I have checked this with a tiny Go program and I can confirm this is not needed at all since this bit is set by default.
So let's remove it completely, and keep only the `dirMode`

Reply:
Hmm. I'll quote again, more, from the golang [docs](https://golang.org/pkg/os/#FileMode):

> A FileMode represents a file's mode and permission bits. The bits have the same definition on all systems, so that information about files can be moved from one system to another portably. Not all bits apply to all systems. The only required bit is ModeDir for directories.

which implies to me, for future system portability, we should have the bit set. It is doing no harm, so I vote to leave it to be 'correct' according to the golang docs - no ?

Reply:
Oh yeah I've missed that sorry... I agree it looks safer to keep this in. Then please move this inside the var `dirMode` !

Reply:
np. I'll rework and merge in the tests to the commit as well...